### PR TITLE
CANVAS-132 - Order Admin Panel listings

### PR DESCRIPTION
**Pull request by chern**

### CODE REVIEW

File: src/server/services/BruincastServices.js [hunk 1]
```diff
@@ -166,6 +167,17 @@ class BruincastServices {
         delete entry._id;
         delete entry.timestamp;
       }
+
+      // Sort each course's media listings by date. If date is the same, then sort by title.
+      courseMedia.listings.sort(function(listing1, listing2) {
+        if (listing1.date < listing2.date) return -1;
+        if (listing2.date < listing1.date) return 1;
+        if (listing1.date === listing2.date && listing1.title < listing2.title)
```
Comment (maxpluez) [inline src/server/services/BruincastServices.js:175]: I don't think you need date1===date2? If we reach this line we already have date1===date2 for sure right?

File: src/server/services/MusicresServices.js [hunk 2]
```diff
@@ -48,6 +48,30 @@ class MusicresServices {
...
+   * @returns {Array}   Formatted media entries.
+   */
+  static formatAllMusicReservesListings(media) {
+    for (const courseMedia of media) {
+      // Sort media for a course group by title.
+      // The sort callback compares SRS first so that items in the "null" course are kept together.
+      courseMedia.listings.sort(function(listing1, listing2) {
+        if (listing1.srs < listing2.srs) return -1;
+        if (listing2.srs < listing1.srs) return 1;
+        if (listing1.srs === listing2.srs) {
```
Comment (maxpluez) [inline src/server/services/MusicresServices.js:64]: Same as above

File: src/server/services/VideoresServices.js [hunk 3]
```diff
@@ -62,6 +62,30 @@ class VideoresServices {
...
+   * @returns {Array}   Formatted media entries.
+   */
+  static formatAllVideoReservesListings(media) {
+    for (const courseMedia of media) {
+      // Sort media for a course group by videoTitle.
+      // The sort callback compares SRS first so that items in the "null" course are kept together.
+      courseMedia.listings.sort(function(listing1, listing2) {
+        if (listing1.srs < listing2.srs) return -1;
+        if (listing2.srs < listing1.srs) return 1;
+        if (listing1.srs === listing2.srs) {
```
Comment (maxpluez) [inline src/server/services/VideoresServices.js:78]: Same as above

File: src/server/services/BruincastServices.js (hunk 1 above)
Comment (chern) [reply src/server/services/BruincastServices.js:175]: Oh yeah, that's true, whoops

File: src/server/services/BruincastServices.js [hunk 4]
```diff
@@ -166,6 +167,15 @@ class BruincastServices {
         delete entry._id;
         delete entry.timestamp;
       }
+
+      // Sort each course's media listings by date. If date is the same, then sort by title.
+      courseMedia.listings.sort(function(listing1, listing2) {
+        if (listing1.date < listing2.date) return -1;
```
Comment (rlorenzo) [inline src/server/services/BruincastServices.js:173]: @chern This cannot be sorted on the database query?

Comment (chern) [reply src/server/services/BruincastServices.js]: @rlorenzo Good point, hadn't thought of that; the latest commit now sorts the listings during db aggregation

**Comment by maxpluez** [comment]
Just tested. Tracks within albums of digital audio reserve are not sorted. Do we need to sort these?
They are not sorted in Digital audio reserve tab itself, and the tracks are under the items fields of albums and might be already sorted in some logical ways.

**Comment by maxpluez** [comment]
> Just tested. Tracks within albums of digital audio reserve are not sorted. Do we need to sort these?
> They are not sorted in Digital audio reserve tab itself, and the tracks are under the items fields of albums and might be already sorted in some logical ways.

If we don't need to sort tracks then this ticket is good to be merged.

**Comment by rlorenzo** [comment]
@chern @maxpluez The tracks should be sorted by track order

**Comment by chern** [comment]
@rlorenzo It looks like the tracks are already ordered by track number in the db (based on how update-musicres reads in the data), so is sorting the tracks in code necessary?

**Comment by rlorenzo** [comment]
@chern  just responding that @maxpluez  said they were unordered.

**Pull request merged**